packages/types/src/type/ExtrinsicEra.ts

```diff
@@ -2,33 +2,189 @@
+    super({
+      period : U8, phase : U8
+    }, MortalEra.decodeMortalEra(value));
+  }
+
+  private static decodeMortalEra (value: MortalMethod | Uint8Array): MortalEraValue {
+    if (isU8a(value)) {
+      const first = u8aToBn(value.subarray(0, 1)).toNumber();
+      let second = u8aToBn(value.subarray(1, 2)).toNumber();
+      const encoded: number = first + (second << 8);
+      const period = 2 << (encoded % (1 << 4));
+      const quantizeFactor = Math.max(period >> 12, 1);
+      let phase = (encoded >> 4) * quantizeFactor;
+      if (period >= 4 && phase < period) {
+        return [new U8(period), new U8(phase)];
+      }
+      throw new Error('Invalid data passed to Mortal era');
+    } else if (isObject(value)) {
+      const { current } = value;
+      const { period } = value;
+      let calPeriod = Math.pow(2, Math.ceil(Math.log2(period)));
+      calPeriod = Math.min(Math.max(calPeriod, 4), 1 << 16);
+      const phase = current % calPeriod;
+      const quantizeFactor = Math.max(calPeriod >> 12, 1);
+      const quantizedPhase = phase / quantizeFactor * quantizeFactor;
+      return [new U8(calPeriod), new U8(quantizedPhase)];
     }
+    throw new Error('Invalid data passed to Mortal era');
+  }
+  /**
+   * @description The period of this Mortal wraps as a [[U8]]
+   */
+  get period (): U8 {
+    return this[0] as U8;
+  }
+
+  /**
+   * @description The phase of this Mortal wraps as a [[U8]]
+   */
+  get phase (): U8 {
+    return this[1] as U8;
+  }
+
+  /**
+   * @description Encodes the value as a Uint8Array as per the parity-codec specifications
+   * @param isBare true when the value has none of the type-specific prefixes (internal)
+   */
+  toU8a (isBare?: boolean): Uint8Array {
+    const period = this.period.toNumber();
+    const phase = this.phase.toNumber();
+    const quantizeFactor = Math.max(period >> 12, 1);
+    const trailingZeros = this.getTrailingZeros(period);
+    const encoded = Math.min(15, Math.max(1, trailingZeros - 1)) + (((phase / quantizeFactor) << 4));
+    const first = encoded >> 8;
+    const second = encoded & 0xff;
+    return new Uint8Array([second, first]);
+  }
 
-    return new Uint8Array([0]);
+  /**
+   * @description Get the block number of the start of the era whose properties this object describes that `current` belongs to.
+   */
+  birth (current: number) {
+    return Math.floor((Math.max(current,this.phase.toNumber()) - this.phase.toNumber()) / this.period.toNumber()) * this.period.toNumber() + this.phase.toNumber();
+  }
+
+  /**
+   * @description Get the block number of the first block at which the era has ended.
+   */
+  death (current: number) {
+    return this.birth(current) + this.period.toNumber();
+  }
+
+  /**
+   * @description convert the number to binary and get the trailing zero's.
+   */
+  private getTrailingZeros (period: number) {
+    const binary = period.toString(2);
+    let index = 0;
+    while (binary[binary.length - 1 - index] === '0') {
```

Comment:
(nit) Spacing before and after the loop.

Reply:
done